cmake/fw_zip.cmake

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+function(generate_dfu_zip)
+  set(oneValueArgs OUTPUT TYPE TARGET)
+  set(multiValueArgs BIN_FILES SCRIPT_PARAMS)
+  cmake_parse_arguments(GENZIP "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  if (NOT(
+    GENZIP_BIN_FILES AND
+    GENZIP_SCRIPT_PARAMS AND
+    GENZIP_OUTPUT AND
+    GENZIP_TYPE AND
+    GENZIP_TARGET
+    ))
+    message(FATAL_ERROR "Missing required param")
```

Comment:
You could be a little nicer
```suggestion
   message(FATAL_ERROR "Missing required param, generate_dfu_zip() requires TARGET, OUTPUT, BIN_FILES, TYPE, SCRIPT_PARAMS to be set.")
```
but your improvement is a the minimum required, so approving.

Reply:
I'm not that nice. Anybody spotting the error message will browse the file generating it, and this avoids having to update the list in the error message if more params suddenly are required.

Reply:
>  if params suddenly are required.

if more params are added, you anyway need to update `cmake_parse_arguments()` as well as the `if()` statement 😛

Reply:
yes, but not the error message :)